PURL-TYPES.rst

```diff
@@ -356,6 +356,16 @@ huggingface
       pkg:huggingface/microsoft/deberta-v3-base@559062ad13d311b87b2c455e67dcd5f1c8f65111?repository_url=https://hub-ci.huggingface.co
 
 
+luarock
```

Comment:
the tool is called `luarocks` (plural), so the singular version seems a bit weird to me.

Reply:
My thought process is that if the ecosystem was called for example `RandomBinaries` then the Purl would `pkg:RandomBinary/SomeRandomBinary`

Reply:
LuaRocks is actually singular despite ending with an s. That's the reason LuaRock is weird. LuaRocks (singular) is the name of the package manager which installs rocks (plural of rock).

It's the same as RubyGems. RubyGems (singular) is the name of the package manager which installs gems (plural of gem). PURL uses `pkg:gem`.

Reply:
So there are examples for both cases then where a single packager is referred to in singular or plural. The reason we named it LuaRock is that rock might conflict so a rock for Lua.

Ultimately, it doesn't really matter to me, whatever the community choses is fine with me

Reply:
If we're worried about collisions it should probably be `luarocks` like how it is with `cocoapods`. LuaRock is not the singular form of LuaRocks. Package names are lowercase so I guess `luarock` could be read as "Lua rock", but it seems close enough to LuaRocks to cause confusion.

Reply:
Not natively English, but the common way to refer to them is `luarocks` for the tool, and a `rock` or `rockspec` for the packages and mainfests. If you use `luarock` anywhere, users will be confused probably whether you mean the tool/group, or an individual rock.

So to me `pkg:luarocks/penlight` would make most sense.